wagtail/contrib/search_promotions/models.py

```diff
@@ -91,26 +143,71 @@ class Meta:
 
 
 class SearchPromotion(models.Model):
+    """
+    Model for search promotion.
+
+    The promotion is for either a page or external url, not both.
+
+    Attributes
+       - query (models.ForeignKey): Matching query object.
+       - page (wagtail.models.Page): Matching page, if any.
+       - external_link_url (models.URLField): Matching external url, if any.
+       - external_link_title (models.CharField) Title text, if external url.
+       - description (models.TextField): Text description of search promotion.
+       - sort_order (int): Ordinal of search promotion.
```

Comment:
```
```

Keep the main docstring entry about the page or external URL, that's useful. listing each attribute is not helpful and will easily get out of sync with the actual code.

This kind of documentation can be auto-generated if we need it for user facing docs. For Developers, they can look at the model.

We do not use this pattern anywhere else in Wagtail, so no need to add it now unless there is a direction from the development guidelines to do so.